cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -552,6 +553,28 @@ private Optional<EmittedRecords> tryRelationalTransform(
 
     if (plugin instanceof RelationalTransform) {
       RelationalTransform transform = (RelationalTransform) plugin;
+
+      if (inputDataCollections.size() > 1 && transform.requireUnionInputs()) {
+        Iterator<String> keys = inputDataCollections.keySet().iterator();
+        String firstKey = keys.next();
+        Schema firstSchema = stageSpec.getInputSchemas().get(firstKey);
+        SparkCollection<Object> first = inputDataCollections.get(firstKey);
+
+        while (keys.hasNext()) {
+          String currKey = keys.next();
+          //Validate schema is Same for all
+          if (firstSchema.equals(stageSpec.getInputSchemas().get(currKey))) {
+            first = first.union(inputDataCollections.get(currKey));
```

Comment:
Please raise a separate JIRA to push this down. Currently it will pull things in. We don't need to address it right away in this PR, but this would defeat all the speedup